kaldb/src/main/java/com/slack/kaldb/elasticsearchApi/ElasticsearchApiService.java

```diff
@@ -115,11 +113,15 @@ private EsSearchResponse doSearch(EsSearchRequest request) throws IOException {
     }
   }
 
-  private HitsMetadata getHits(KaldbSearch.SearchResult searchResult) throws IOException {
+  private HitsMetadata getHits(KaldbSearch.SearchResult searchResult) {
     List<ByteString> hitsByteList = searchResult.getHitsList().asByteStringList();
     List<SearchResponseHit> responseHits = new ArrayList<>(hitsByteList.size());
     for (ByteString bytes : hitsByteList) {
-      responseHits.add(SearchResponseHit.fromByteString(bytes));
+      try {
+        responseHits.add(SearchResponseHit.fromByteString(bytes));
+      } catch (IOException e) {
+        LOG.error("Error processing search hits bytes", e);
```

Comment:
This api can fail all calls and we would still return a 200 error. It would be good catch this case using a metric. Do we have any metrics that provide visibility into these exceptions? If not, let's add a metric for successful and failed requests.

Reply:
I reworked the error handling here to account for the previous errors with deserialization, as well as any other runtime exceptions that aren't currently handled. These now are properly handled, logged, and reported into the span as well as reported with a 500 status on the multisearch request that failed (along with the trace ID).